### ISSUE TITLE
Latest (recency) CoreOS stable image in us-east-1 is outdated

### ISSUE THREAD

**Issue by valer-cara** [poseidon/typhoon#522]
### Environment

* Platform: aws
* OS: container-linux
* Release: ref=v1.15.1
* Terraform: Terraform v0.12.5            
* Plugins: 
+ provider.aws v2.15.0       
+ provider.ct v0.3.2         
+ provider.local v1.3.0      
+ provider.null v2.1.2       
+ provider.template v2.1.2   
+ provider.tls v2.0.1        

### Problem

Neither controller nor workers booting in fresh setup of cluster.
The controller logs show an error related to ignition config not found, URL not provided.

Looking in depth, terraform actually picked an older version of CoreOS as the `most_recent`. I've checked the AMIs and it seems there's been a massive re-issue of AMIs matching the criteria (owner, arch, ..; not sure why).

```
aws ec2 describe-images --output table --owners 595879546273 --filters "Name=name,Values=CoreOS-stable-*" "Name=virtualization-type,Values=hvm" "Name=architecture,Values=x86_64" --query 'sort_by(Images, &CreationDate)[].[ImageId,Name,CreationDate]'
```

![Screenshot_2019-07-31_22:59:38_1703x734_scrot](https://user-images.githubusercontent.com/2016345/62244144-7d267380-b3e7-11e9-837c-a55741aa66d3.png)

> This massive all-version republish is pretty odd, wonder why..

### Desired Behavior

Have the cluster successfully boot.

Or have the ability to pass in which AMI to use. For now i'll just switch to the Beta channel which seems unaffected by this issue.

~For more reliability, maybe hardcoding AMI versions that are known to work would help?~

### Steps to Reproduce

Just boot an AWS/coreos cluster following the documentation.

**Comment by dghubble** [comment]
I can't reproduce your problem.

Listing images via the aws CLI shows `2135.5.0` as the current CoreOS stable release (correct at time or writing) in the regions I checked (`us-east-2`, `us-west-1`). Planning a Typhoon cluster in those test regions shows the Terraform data resource selects the appropriate AMI (e.g. ami-03aa12465ead76468 in `us-east-2`, which matches https://stable.release.core-os.net/amd64-usr/current/coreos_production_ami_hvm_us-east-2.txt).

```
|  ami-0a7e0ff8d31da1836|  CoreOS-stable-2135.4.0-hvm   |  2019-06-25T00:04:55.000Z  |
|  ami-071f4352a744b29aa|  CoreOS-stable-2135.5.0-hvm   |  2019-07-01T21:04:39.000Z  |
```

If you can cross reference the AMI list you see against published [CoreOS Releases](https://coreos.com/releases/) (click browse images) and find an inaccuracy, you can file it with [coreos/bugs](https://github.com/coreos/bugs) to investigate.

If you can get an isolated example of the aws ami data source picking the wrong AMI (like if its an issue with recency), you have a report to [terraform-provider-aws](https://github.com/terraform-providers/terraform-provider-aws)

AMIs are regional, so if there were some issue its possible its isolated to whatever region you're using. Also, check a few more things. Is your system clock reasonably right? I'm sure terraform's most_recent check is fairly simple and could be tricked.

**Comment by dghubble** [comment]
Wait, found it. Its just `us-east-1` and it does look like a re-publish, specifically to `coreos-stable` as you mentioned. 

Since Typhoon selects AMIs from many sources and any region, I'm inclined to raise this with coreos/bugs and not special case it. Even if no action is taken, the next coreos-stable release will solve it.

Until more is known or a new coreos-stable release is published, I'd say pick a different channel or region.

**Comment by dghubble** [comment]
Looks like they had to re-publish those and its intentional. https://github.com/coreos/bugs/issues/2604

It'll disappear when the next CoreOS stable release is cut. You can pick another channel or region in the mean time or just carry a tiny fork if you really need that specific region for a few days. I'm not planning to mainline a workaround since it'll disappear on its own soon.

**Comment by valer-cara** [comment]
Thanks for investigating! Yes, i'm on `us-east-1`. Agree that this shouldn't be a special case and I don't see a way to tackle these edgy situations using the `aws_ami` resource.

Wonder if an additional override of the AMI would be helpful: if `os_image` matches `ami-*`, use it verbatim, otherwise use the existing logic.

Regardless, I think this issue can be closed. Fix is -> change os_image to `coreos-beta`.

**Comment by dghubble** [comment]
Should be resolved already thanks to the CoreOS Stable 2135.6.0 release!
